src/modules/idle_inhibitor.cpp

```diff
@@ -19,37 +25,60 @@ waybar::modules::IdleInhibitor::~IdleInhibitor() {
     zwp_idle_inhibitor_v1_destroy(idle_inhibitor_);
     idle_inhibitor_ = nullptr;
   }
+
+  // Remove this from the modules list
+  waybar::modules::IdleInhibitor::modules.remove(this);
+
   if (pid_ != -1) {
     kill(-pid_, 9);
     pid_ = -1;
   }
 }
 
 auto waybar::modules::IdleInhibitor::update() -> void {
+  // Check status
+  std::string status = waybar::modules::IdleInhibitor::status;
+  if (status == "activated") {
+    if (idle_inhibitor_ == nullptr) {
+      idle_inhibitor_ = zwp_idle_inhibit_manager_v1_create_inhibitor(
+        waybar::Client::inst()->idle_inhibit_manager, bar_.surface);
+    }
+  } else {
+    if (idle_inhibitor_ != nullptr) {
+      zwp_idle_inhibitor_v1_destroy(idle_inhibitor_);
+      idle_inhibitor_ = nullptr;
+    }
+  }
+
   label_.set_markup(
-      fmt::format(format_, fmt::arg("status", status_), fmt::arg("icon", getIcon(0, status_))));
-  label_.get_style_context()->add_class(status_);
+      fmt::format(format_, fmt::arg("status", status), fmt::arg("icon", getIcon(0, status))));
+  label_.get_style_context()->add_class(status);
   if (tooltipEnabled()) {
-    label_.set_tooltip_text(status_);
+    label_.set_tooltip_text(status);
   }
   // Call parent update
   ALabel::update();
 }
 
 bool waybar::modules::IdleInhibitor::handleToggle(GdkEventButton* const& e) {
   if (e->button == 1) {
-    label_.get_style_context()->remove_class(status_);
-    if (idle_inhibitor_ != nullptr) {
-      zwp_idle_inhibitor_v1_destroy(idle_inhibitor_);
-      idle_inhibitor_ = nullptr;
-      status_ = "deactivated";
+    std::string status = waybar::modules::IdleInhibitor::status;
+    label_.get_style_context()->remove_class(status);
+    if (status == "activated") {
+      status = "deactivated";
     } else {
-      idle_inhibitor_ = zwp_idle_inhibit_manager_v1_create_inhibitor(
-          waybar::Client::inst()->idle_inhibit_manager, bar_.surface);
-      status_ = "activated";
+      status = "activated";
     }
-    click_param = status_;
+    waybar::modules::IdleInhibitor::status = status;
   }
+
+  // Make all other idle inhibitor modules update
+  for (auto const& module : waybar::modules::IdleInhibitor::modules) {
+    if (module != this) {
+      module->update();
```

Comment:
Current module also needs to be updated.

Reply:
Current module is updated after its handleToggle is run, I'm not sure how/where but it is definitely triggered.

Reply:
Right, `AModule::handleToggle` already triggers update by calling `dp.emit`. Forgot about that.